plugin/mlsdbprovider.cpp

```diff
@@ -269,21 +269,29 @@ void MlsdbProvider::calculatePositionAndEmitLocation()
     qCDebug(lcGeoclueMlsdbPosition) << "have" << m_cellWatcher->cells().size() << "neighbouring cells";
     QList<CellPositioningData> cells;
     quint32 maxNeighborSignalStrength = 1;
-    QSet<quint32> seenCellIds;
+    QSet<quint64> seenCellIds;
     Q_FOREACH (const QSharedPointer<QOfonoExtCell> &c, m_cellWatcher->cells()) {
         CellPositioningData cell;
+        MlsdbCellType cellType = MLSDB_CELL_TYPE_LTE;
+        quint32 locationCode = 0;
+        quint32 cellId = 0;
         if (c->cid() != -1 && c->cid() != 0) {
             // gsm / wcdma
-            cell.cellId = c->cid();
+            cellType = MLSDB_CELL_TYPE_GSM_WCDMA;
+            locationCode = static_cast<quint32>(c->lac());
+            cellId = static_cast<quint32>(c->cid());
         } else if (c->ci() != -1 && c->ci() != 0) {
             // lte
-            cell.cellId = c->ci();
+            cellType = MLSDB_CELL_TYPE_LTE;
+            locationCode = static_cast<quint32>(c->tac());
+            cellId = static_cast<quint32>(c->ci());
         } else {
-            qCDebug(lcGeoclueMlsdbPosition) << "ignoring neighbour cell with no cell id:" << c->type() << c->mcc() << c->mnc();
+            qCDebug(lcGeoclueMlsdbPosition) << "ignoring neighbour cell with no cell id:" << c->type() << c->mcc() << c->mnc() << c->lac() << c->tac();
```

Comment:
Would be good to add here also what the values are to make easier to debug. something like 

qCDebug(lcGeoclueMlsdbPosition) << "ignoring neighbour cell with no cell id. type:" << c->type() << ", mcc: " << c->mcc() << ", mnc: " << c->mnc();